install/install.php

```diff
@@ -365,84 +381,96 @@ function next_form() {
       Html::closeForm();
    }
 
-   //Check if the port is in url
    $hostport = explode(":", $host);
+   $driver   = 'mysql';
    if (count($hostport) < 2) {
-      $link = new mysqli($hostport[0], $user, $password);
+      // Host
+      $dsn = "$driver:host=$host";
+   } else if (intval($hostport[1])>0) {
+       // Host:port
+       $dsn = "$driver:host={$hostport[0]}:{$hostport[1]}";
    } else {
-      $link = new mysqli($hostport[0], $user, $password, '', $hostport[1]);
+       // :Socket
+       $dsn = "$driver:unix_socket={$hostport[1]}";
    }
 
-   $databasename    = $link->real_escape_string($databasename);
-   $newdatabasename = $link->real_escape_string($newdatabasename);
+   $connected = false;
+   try {
+      $link = new PDO(
+         "$dsn;charset=utf8",
+         $user,
+         $password
+      );
+      $connected = true;
+   } catch (PDOException $e) {
+      echo "<p>".__("Can't connect to the database")."\n <br>".
+           sprintf(__('The server answered: %s'), $e->getMessage())."</p>";
+   }
+   $link->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
 
-   if (!empty($databasename)) { // use db already created
-      $DB_selected = $link->select_db($databasename);
+   $databasename = trim($link->quote($databasename), "'");
```

Comment:
Using quote will fail if table name contains a `.`.

Reply:
I've dropped quoting and used only quoteName in the CREATE query.